src/reducers/SensorDetailReducer.js

```diff
@@ -0,0 +1,80 @@
+import { SENSOR_DETAIL_ACTIONS } from '../actions/SensorDetailActions';
+import { ROUTES } from '../navigation/index';
+import { SECONDS } from '../utilities/constants';
+
+const getState = sensor => ({
+  name: sensor.name,
+  code: sensor.location?.code,
+  macAddress: sensor?.macAddress,
+  logInterval: sensor?.logInterval,
+  HOT_CUMULATIVE: { duration: 0, threshold: 99, temperature: 2 },
+  HOT_CONSECUTIVE: { duration: 0, threshold: 99, temperature: 2 },
+  COLD_CUMULATIVE: { duration: 0, threshold: 99, temperature: 2 },
+  COLD_CONSECUTIVE: { duration: 0, threshold: 99, temperature: 2 },
+  batteryLevel: sensor.batteryLevel,
+  lastSyncDate: new Date(),
+});
+
+export const initialState = () => ({
+  name: '',
+  code: '',
+  macAddress: '',
+  logInterval: 300,
+  HOT_CUMULATIVE: { duration: 0, threshold: 99, temperature: 2 },
+  HOT_CONSECUTIVE: { duration: 0, threshold: 99, temperature: 2 },
+  COLD_CUMULATIVE: { duration: 0, threshold: 99, temperature: 2 },
+  COLD_CONSECUTIVE: { duration: 0, threshold: 99, temperature: 2 },
+  batteryLevel: 99,
+  lastSyncDate: new Date(),
```

Comment:
This might need to live somewhere else if it's going to be used in many places as the last sync date